Lib/test/test_socket.py

```diff
@@ -4067,8 +4073,44 @@ def testSetBlocking(self):
         # Testing whether set blocking works
         self.serv.setblocking(True)
         self.assertIsNone(self.serv.gettimeout())
+        self.assertTrue(self.serv.getblocking())
+        if fcntl:
+            self.assertTrue(_is_fd_in_blocking_mode(self.serv))
+
         self.serv.setblocking(False)
         self.assertEqual(self.serv.gettimeout(), 0.0)
+        self.assertFalse(self.serv.getblocking())
+        if fcntl:
+            self.assertFalse(_is_fd_in_blocking_mode(self.serv))
```

Comment:
You may write an helper function which makes sure that timeout, blocking and SOCK_NONBLOCK flag are consistent.

Something like:

```
def check_timeout(sock, timeout):
  assert sock.gettimeout() == timeout
  assert sock.getblocking() == (timeout == 0.0)
  assert _is_fd_in_blocking_mode(sock) == (timeout != 0.0)
```

(I'm not sure of these assertions :-))